internal/injector/builtin/yaml/datastreams/confluentinc_kafka_v2.yml

```diff
@@ -0,0 +1,44 @@
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2023-present Datadog, Inc.
+---
+# yaml-language-server: $schema=../../../../../docs/static/schema.json
+meta:
+  name: github.com/confluentinc/confluent-kafka-go/v2/kafka
+  description: Sarama is a Go library for Apache Kafka
+  icon: fast-forward
+
+aspects:
+  - id: Wrap kafka.NewConsumer
+    join-point:
+      function-call: github.com/confluentinc/confluent-kafka-go/v2/kafka.NewConsumer
+    advice:
+      - wrap-expression:
+          imports:
+            kafkatrace: gopkg.in/DataDog/dd-trace-go.v1/contrib/confluentinc/confluent-kafka-go/kafka.v2
+            kafka: github.com/confluentinc/confluent-kafka-go/v2/kafka
+          template: |-
+            func(c *kafka.Consumer, err error) (*kafkatrace.Consumer, error) {
+              var wc *kafkatrace.Consumer
+              if c != nil {
+                wc = kafkatrace.WrapConsumer(c)
+              }
+              return wc, err
+            }({{ . }})
+  - id: Wrap kafka.NewProducer
+    join-point:
+      function-call: github.com/confluentinc/confluent-kafka-go/v2/kafka.NewProducer
+    advice:
+      - wrap-expression:
+          imports:
+            kafkatrace: gopkg.in/DataDog/dd-trace-go.v1/contrib/confluentinc/confluent-kafka-go/kafka.v2
+            kafka: github.com/confluentinc/confluent-kafka-go/v2/kafka
+          template: |-
+            func(p *kafka.Producer, err error) (*kafkatrace.Producer, error) {
```

Comment:
I don't think this is a safe replacement to do, as we originally have a `*kafka.Producer` and we return a `*kafkatrace.Producer`

if the user has code like this, it would break:

```
var (
	producer *kafka.Producer
	err error
)

producer = // this replacement
```

Also, if there are any type assertions in the users code, these would break.

I think the only safe approach here is to instrument the library code itself, similar as I did here: https://github.com/DataDog/orchestrion/pull/266

cc @RomainMuller @eliottness please confirm if this is correct 🙇

Reply:
I can confirm. There is the same issue with gorilla/mux